app/controllers/posts_controller.rb

```diff
@@ -2,12 +2,27 @@ class PostsController < ApplicationController
   def index
     @posts = Post.where(author_id: params[:user_id])
     @posts = @posts.order(created_at: :desc)
+    @user = User.find(params[:user_id])
   end
 
   def show
     @post = Post.find(params[:id])
   end
 
+  def destroy
+    if current_user.role == 'admin'
+      @post = Post.find(params[:id])
+      author = @post.author
+      @post.destroy
+      redirect_to user_posts_path(author)
+    else
+      @user = current_user
+      @post = Post.find(params[:id])
+      @post.destroy
+      redirect_to user_posts_path(@user)
+    end
+  end
+
```

Comment:
- [ ] Nice work on implementing the destroy logic,however the authorization should bemade with cancan, I can see you are manualy checking if the user the admin. You simply need to add the midlware `load_and_authorize_resource` which will be checking authomaticaly if the user is the Admin.

```rb
class PostsController < ApplicationController 
  `load_and_authorize_resource`
  def index
  end 
  ...
  ....
end

```

Reply:
Done Bro!